### PR TITLE
suggested edits for the webhook label fixes

### PR DESCRIPTION
**Pull request by ca-scribner**
* replaces kubectl call with lightkube
* splits tests into separate build_and_deploy and seldon_deployment (eg: job) tests for easier debugging of tests (this way if debugging the sample seldon job, the tests can run with `-k test_seldon_deployment` and wont rebuild/deploy the charm every time)
* runs everything in the test's juju model/namespace

### CODE REVIEW

**Review by DomFleischmann — APPROVED** [review]
LGTM but would like @knkski to take a look as I'm missing context.

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -41,6 +34,14 @@ async def test_build_and_deploy(ops_test: OpsTest):
     )
     assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
 
+
+async def test_seldon_deployment(ops_test: OpsTest):
+    namespace = ops_test.model_name
+    client = Client()
+
+    this_ns = client.get(res=Namespace, name=namespace)
+    this_ns.metadata.labels.update({"serving.kubeflow.org/inferenceservice": "enabled"})
```
Comment (knkski) [inline tests/integration/test_charm.py:43]: I don't think this will set the label, it will just update the in-memory Python object

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -1,8 +1,10 @@
 name: CI
 
 on:
-  - push
-  - pull_request
+  push:
```
Comment (knkski) [inline .github/workflows/ci.yaml:4]: This should probably be in a separate PR

File: tests/integration/test_charm.py (hunk 1 above)
Comment (ca-scribner) [reply tests/integration/test_charm.py:43]: Good catch, there was a missing call to `client.patch()`

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (ca-scribner) [reply .github/workflows/ci.yaml]: Agreed.  Moved to #16
